x-pack/plugins/security_solution/public/management/pages/policy/view/policy_hooks.ts

```diff
@@ -28,3 +40,60 @@ export function usePolicyDetailsSelector<TSelected>(
     )
   );
 }
+
+export type NavigationCallback = (
+  ...args: Parameters<Parameters<typeof useCallback>[0]>
+) => Partial<PolicyDetailsArtifactsPageLocation>;
+
+export function usePolicyDetailsNavigateCallback() {
```

Comment:
ohhhh - this looks like a good approach to addressing the original issue that is currently found in the Trusted apps pages. 👍